backend/app/checkpoint.py

```diff
@@ -9,6 +11,23 @@
 from app.lifespan import get_pg_pool
 
 
+class PostgresUnpickler(pickle.Unpickler):
```

Comment:
It might be a newbie question/comment, but I tend to shy away from storing language-specific specific serialization formats. Would it be better to use a data format that supports schema migrations?

Reply:
Sure we can migrate to something else, if there is a different serialization format that supports arbitrary python objects?